### DISCUSSION TITLE
.net 8 MAUI confusion

### DISCUSSION THREAD

**Discussion by sisaacks** [roubachof/MetroLog#33] [Q&A]
So I looked through the github examples for setting up .net MAUI configuration.  
When I get to a class to use the logger there are issues

The first thing I notices is MetroLog.MAUI is not the the using statement its using MetroLog but the examples are showing MetroLog.MAUI

Second thing I noticed is to do the following ILogger<MyClass> _logger  I have to add the extension using Microsoft.Extensions.Logging;
and Using MetroLog is now unused.  So I am absolutely confused if I am using MetroLogger or Microsoft.Extensions.Logging?

I am not sure if I have set something up wrong, but I have watched a video from Gerald Versluis and I looked through github examples.  I am not sure what I am doing wrong.

**Comment by roubachof** [comment]
You can you use either standard or Microsoft extensions.
If ytou use standard you instanciate your Logger with logger manager static factory, if you use MS you inject in the constructor.
If you use standard you will write using MetroLog, is you use MS you will write using MS.Ext.Logging

**Reply by sisaacks** [reply]
If I understand what you typed above:
If I use the standard way to instantiate the  Logger (using the static method) and logging will be done with MetroLog
If I use DI, then I will be using the Microsoft Extensions for Logging, not MetroLog
IF that is the case why are the sample in Github incorrect for .net MAUI?

**Reply by sisaacks** [reply]
@roubachof  this -->  static readonly ILogger Logger = LoggerFactory.GetLogger(nameof(MainPage)); throws an error when running.  
I am still 100% confused as to how to implement this.  nothing seems to work like the documentation states.  The error states the logger has not been initialized.......I have a .net MAUI project and I set up the configuration in the MauiProgram.cs.  So I am guessing I need to remove the .net MAUI setup (as your example shows) and create a LoggerFactory configuration?

**Reply by roubachof** [reply]
just follow this: https://github.com/roubachof/MetroLog?tab=readme-ov-file#standard-configuration
if you want the stabndard way with LoggerFactory

**Reply by sisaacks** [reply]
@roubachof I did...but I am also using .net MAUI, so I set up my logging per your example....which is incorrect if you want to use standard logging.  I didnt realize that at first because the documentation does not explain that.  Where should the logging configuration file be and when should it be set up?  Can you do this from the MauiProgram.cs file?

**Reply by roubachof** [reply]
yes the doc explains that, it's either standard configuration either ms extensions configuration

**Reply by sisaacks** [reply]
@roubachof  If its written there its not very clear, at least not to me.  I have set up my logging using using Microsoft.Extensions.Logging passing in the ILogger via DI.  I do have Base classes that need a logger.  How do I capture the logger without using DI?  Being that base classes have no constructor.

**Reply by roubachof** [reply]
you can't, if you using MS Extensions it's injection. Or make some search about MS Extenstions without using DI. If you use MS Extensions metrolog is totally abstracted in the MS Extensions logging.

**Reply by sisaacks** [reply]
@roubachof I rang a google search after I typed the question to you, the answer is the make your base class and abstract class and pass the logger down through the constructor.  I made a sample test app to ensure it worked.

**Reply by roubachof** [reply]
and ? so do that ! I'm sorry but I am so confused with what you are struggling :)

**Reply by sisaacks** [reply]
@roubachof Interesting

**Reply by roubachof** [reply]
I have the feeling you are trolling me for some reason so I will stop to participate in this discussion
